include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -11,125 +11,129 @@
 #pragma once
 
 #include <eve/as.hpp>
+#include <eve/concept/memory.hpp>
 #include <eve/concept/vectorizable.hpp>
 #include <eve/detail/abi.hpp>
-#include <eve/detail/meta.hpp>
-#include <eve/memory/aligned_ptr.hpp>
+#include <eve/detail/category.hpp>
 
 namespace eve::detail
 {
   //================================================================================================
-  // 128 bits loads
+  // Regular loads
   //================================================================================================
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, eve::x86_128_ const &, T const* p)
+  template<scalar_pointer Ptr, typename T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, ABI const &, Ptr p)
+  requires( std::same_as<T, std::remove_cvref_t<decltype(*p)>> )
   {
-    if constexpr( N::value * sizeof(T) == x86_128_::bytes )
+    constexpr auto cat = categorize<wide<T, N>>();
+    constexpr bool isfull256 = N::value*sizeof(T) == x86_256_::bytes;
+    constexpr bool isfull128 = N::value*sizeof(T) == x86_128_::bytes;
+
+    if constexpr( !std::is_pointer_v<Ptr> )
```

Comment:
I think this should probably be a proper check for `aligned_ptr<, sizeof(T) * N>`
Otherwise this will break.

Also - I believe that this will eat a `unique_ptr` or some other nonsense.

Reply:
I thought of the `unique_ptr` situation. I can refine the concept to be T* or aligned_ptr<T,N>.
The concept makes the code far easier to work with and more compact.

Reply:
Fixed to specifically test for is_pointer + aligned_ptr